onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -1346,66 +1348,102 @@ TEST(InferenceSessionTests, TestCopyToFromDevices) {
+    SessionOptions so;
+    so.session_logid = "InferenceSessionTests.TestL1AndL2Transformers";
+    so.graph_optimization_level = static_cast<TransformerLevel>(i);
+    InferenceSession session_object{so, &DefaultLoggingManager()};
 
-  std::shared_ptr<Model> p_model;
-  ASSERT_TRUE(Model::Load(model_uri, p_model).IsOK());
+    // Create and register dummy graph transformer
+    auto dummy_transformer_unique_ptr = std::make_unique<DummyGraphTransformer>("DummyTransformer");
+    const auto* dummy_transformer = dummy_transformer_unique_ptr.get();
+    session_object.RegisterGraphTransformer(std::move(dummy_transformer_unique_ptr));
 
-  Status st = session_object.Initialize();
-  ASSERT_TRUE(st.IsOK()) << st;
+    session_object.Load(model_uri);
+    ASSERT_TRUE(session_object.Initialize().IsOK());
+
+    // Validate transformer was called after Session.Initialize
+    ASSERT_TRUE(dummy_transformer->IsTransformerInvoked());
+  }
 }
 
+// This test validates session initialize is successful when all the pre-defined
+// L1 and L2 transformers are enabled.
 TEST(InferenceSessionTests, TestL1AndL2Transformers) {
+  // Models which cover all transformers.
+  std::vector<std::string> test_model_uris = {"testdata/transform/fusion/fuse-conv-bn-mul-add-unsqueeze.onnx",
+                                              "testdata/transform/abs-id-max.onnx",
+                                              "testdata/transform/slice-elim.onnx",
+                                              "testdata/transform/matmul_add_fusion/2Input/model.onnx",
+                                              "testdata/transform/matmul_add_fusion/3Input/gemm_relu.onnx",
+                                              "testdata/transform/fusion/fuse-conv-bn-add-mul-float16.onnx"};
+
+  for (const auto& model_uri : test_model_uris) {
+    SessionOptions so;
+    so.session_logid = "InferenceSessionTests.TestL1AndL2Transformers";
+    so.graph_optimization_level = TransformerLevel::Level2;
+    InferenceSession session_object{so, &DefaultLoggingManager()};
+    ASSERT_TRUE(session_object.Load(model_uri).IsOK());
+    ASSERT_TRUE(session_object.Initialize().IsOK());
+  }
+}
+
+// This test validates AddCustomTransformerList api.
+TEST(InferenceSessionTests, TestCustomTransformers) {
   string model_uri = "testdata/transform/fusion/fuse-conv-bn-mul-add-unsqueeze.onnx";
 
   SessionOptions so;
   so.session_logid = "InferenceSessionTests.TestL1AndL2Transformers";
   so.graph_optimization_level = TransformerLevel::Level2;
   InferenceSession session_object{so, &DefaultLoggingManager()};
-  ASSERT_TRUE(session_object.Load(model_uri).IsOK());
-
-  std::shared_ptr<Model> p_model;
-  ASSERT_TRUE(Model::Load(model_uri, p_model).IsOK());  
+  ASSERT_TRUE(session_object.AddCustomTransformerList({"EliminateIdentity", "ConvAddFusion", "EliminateUnsqueeze"}).IsOK());
 
+  ASSERT_TRUE(session_object.Load(model_uri).IsOK());
   ASSERT_TRUE(session_object.Initialize().IsOK());
 }
```

Comment:
Shouldn't we test something post-initialize to make sure things in the custom transformer list actually ran? Right now it seems to just test that nothing broke and not that it did what was expected. You could just use a dummy transformer that tracks if it's called.